File: services/worker/internal/models/activity_log.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/datatypes"
)

type ActivityLog struct {
	ID         uuid.UUID      `gorm:"type:uuid;primaryKey;default:gen_random_uuid()" json:"id"`
	Action     string         `gorm:"not null" json:"action"`
	EntityType string         `gorm:"not null" json:"entityType"`
	EntityID   *uuid.UUID     `gorm:"type:uuid" json:"entityId"`
	Details    datatypes.JSON `json:"details"`
	CreatedAt  time.Time      `gorm:"default:now()" json:"createdAt"`
}

File: services/core/src/db/schema/activityLog.ts
import { pgTable, uuid, varchar, timestamp, jsonb } from "drizzle-orm/pg-core";

export const activityLogs = pgTable("activity_logs", {
  id: uuid("id").primaryKey().defaultRandom(),
  action: varchar("action", { length: 100 }).notNull(),
  entityType: varchar("entity_type", { length: 100 }).notNull(),
  entityId: uuid("entity_id"),
  details: jsonb("details"),
  createdAt: timestamp("created_at").defaultNow().notNull(),
});
